### ISSUE TITLE
On demand detection without Bookmarklet

### ISSUE THREAD

**Issue by func0der** [AliasIO/Wappalyzer#1903]
I know this topic has been discussed in #201 and #210, but I am not that happy with the outcome.
And for the sake of issue etiquette, I have opened this new issue instead of commenting on closed ones.

To my point:
I find it very handy to have an addon installed, because it is easier obtained than a bookmarklet.

Though it is super great, that there is a bookmarklet for on-demand analysis, I think that it should at least be an advanced feature to trigger Wappalyzer on demand.
If you would want to have the headers for a request (discussed in #201), it should be no problem to reload the page to obtain those. That would work in 95% of the cases and people who disable automatic detection or people who use Wappalyzer in general, should be well aware of the five percent, where that would screw with your form post or whatever.

Maybe there is even a way, that this works by now (WebExtensions), because the bookmarklet shows me, that I am using nginx without reloading the page.

One of the reasons I want this, is that Wappalyzer triggers at least one additional request per site call trying to get information from the robots.txt. That might be great as a research resource, but I do not want that additional request to happen in 98 % of the cases, because it just wastes time and bandwidth.

An on-demand solution would be really appreciated here.

The complexity that this adds to the code should be minimal:
- Config entry to not trigger Wappalyzer automatically on site load
- Config check on site load to (not) trigger analysis
- Button to trigger analysis (i.e. reload the site with a force analyse flag that triggers a one-time run of Wappalyzer)

Maybe I am not seeing something here, but I'd really love this feature, as I love this extension as well :)

**Comment by AliasIO** [comment]
If you disable the option to send anonymous reports, the extension won't make any external calls including robots.txt.

Can you elaborate on other reasons you'd want this feature?

**Comment by func0der** [comment]
I just to spare the extra cpu used during site loads and afterwards for the detection of the different libraries etc.

**Comment by gadcam** [comment]
@func0der As a contributor, I can not see why the bookmarklet would not be the perfect fit for this use case : I am not against this feature but I do not understand why someone would need it.

By the way I do not understand what you mean by
> I find it very handy to have an addon installed, because it is easier obtained than a bookmarklet.

**Comment by AliasIO** [comment]
Closing because of inactivity.

**Comment by func0der** [comment]
@AliasIO
So, YOU are not replying and this ticket is "inactive"?

**Comment by gadcam** [comment]
@func0der I asked some precision about your use case, you did not answered and 2 months after that AliasIO closed the issue. I think he gave you a fair time to express your thoughts but I would be more than pleased if you answer now :)

**Comment by func0der** [comment]
I answered his question and since it is his repository, I thought it would be fair enough for him to react to my answer.

If you want this answered:
> By the way I do not understand what you mean by

>    I find it very handy to have an addon installed, because it is easier obtained than a bookmarklet.

I have searched for about 15 minutes for that bookmarklet. I only search five seconds for the addon in the Firefox addon library.
I would never even try to search for a bookmarklet to do such a thing. I guess, barely anyone would. If there is not extension, there is just no extension. One either gives up or develops it own. And the second is VERY far fetched. Most normal users would consider doing the first.
